apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/service/ConsumerService.java

```diff
@@ -120,10 +120,11 @@ public Consumer createConsumer(Consumer consumer) {
     return consumerRepository.save(consumer);
   }
 
-  public ConsumerToken generateAndSaveConsumerToken(Consumer consumer, Date expires) {
+  public ConsumerToken generateAndSaveConsumerToken(Consumer consumer, Integer rateLimit, Date expires) {
     Preconditions.checkArgument(consumer != null, "Consumer can not be null");
+    Preconditions.checkArgument(rateLimit != null && rateLimit >= 0, "Rate limit must be non-negative");
```

Comment:
This check seems unnecessary, as the logic is already handled in ConsumerController, with default behavior applied in generateConsumerToken.